pkg/operator/ceph/pool/predicate.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2021 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package pool to manage a rook pool.
+package pool
+
+import (
+	"encoding/json"
+	"reflect"
+	"sort"
+
+	"github.com/rook/rook/pkg/operator/ceph/cluster/mon"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+func predicateMonEndpointChanges() predicate.Funcs {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return false
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			cmNew, ok := e.ObjectNew.(*corev1.ConfigMap)
+			if !ok {
+				return false
+			}
+			cmOld, ok := e.ObjectOld.(*corev1.ConfigMap)
+			if !ok {
+				return false
+			}
+			if cmNew.GetName() == mon.EndpointConfigMapName {
+				if wereMonEndpointsUpdated(cmOld.Data, cmNew.Data) {
+					logger.Info("monitor endpoints changed, updating the bootstrap peer token")
+					return true
+				}
+			}
+			return false
+		},
+	}
+}
+
+func wereMonEndpointsUpdated(oldCMData, newCMData map[string]string) bool {
+	// Check the mapping key first
+	if oldMapping, ok := oldCMData["mapping"]; ok {
+		if newMapping, ok := newCMData["mapping"]; ok {
```

Comment:
nit: for both of these checks for old and new mapping, return false immediately to avoid deeper nesting of the code
```suggestion
		if newMapping, ok := newCMData["mapping"]; !ok {
			return false
		}
```

Reply:
Is it really necessary? If `!ok`, the next condition if the other check for existence, and if `!ok` we go to the end of the function and return `nil`.

Reply:
It's personal preference to avoid deeper code blocks, of course either way is functionally the same